lib/galaxy_ext/metadata/set_metadata.py

```diff
@@ -103,7 +103,7 @@ def set_metadata():
             try:
                 line = stringify_dictionary_keys(json.loads(line))
                 if line['type'] == 'dataset':
-                    existing_job_metadata_dict[line['dataset_id']] = line
+                    existing_job_metadata_dict[int(line['dataset_id'])] = line
```

Comment:
Why is the `dataset_id` a string here, that seems weird ...  but good catch. Should we also log the exception on line 109 ?

Reply:
We can't log exceptions in set_metadata.py yet right - might break the job or whatever? Maybe after #7095?

Reply:
Yeah, https://github.com/galaxyproject/galaxy/pull/7095 should do it.